### PR TITLE
feat(dashboard,medusa,ui): Manual gift cards + cleanup

### PR DESCRIPTION
**Pull request by kasperkristensen**
**Dashboard**
- Adds different views for managing manual/custom gift cards (not associated with a product gift card).
- Cleans up several table implementations to use new DataTable component.
- Minor cleanup of translation file.

**Medusa**
- Adds missing query params for list endpoints in the following admin domains: /customers, /customer-groups, /collections, and /gift-cards.

**UI**
- Adds new sizes for Badge component.

**Note for review**
Since this PR contains updates to the translation keys, it touches a lot of files. For the review the parts that are relevant are: the /gift-cards domain of admin, the table overview of collections, customers, and customer groups. And the changes to the list endpoints in the core.

### CODE REVIEW

**Comment by olivermrbl** [comment]
The code cell is off. Should we truncate and add a tooltip in case it gets cut off?

CleanShot 2024-02-12 at 13 53 14

**Comment by olivermrbl** [comment]
The "Original amount" and "Balance" columns should be "Initial balance" and "Current balance" to be consistent with the details page.

**Comment by olivermrbl** [comment]
Should we attempt to render the recipient email on the details page by scanning the metadata?

File: packages/admin-next/dashboard/src/hooks/use-data-table.tsx [hunk 1]
```diff
@@ -10,27 +10,29 @@ import {
...
+  meta?: Record<string, unknown>
   prefix?: string
 }
 
-export const useDataTable = <TData, TValue>({
+export const useDataTable = <TData,>({
   data = [],
   columns,
   count = 0,
   pageSize: _pageSize = 50,
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/hooks/use-data-table.tsx]: suggestion(non-blocking): Default to 20

File: packages/admin-next/dashboard/src/routes/collections/collection-detail/components/collection-product-section/collection-product-section.tsx [hunk 2]
```diff
@@ -155,99 +102,26 @@ export const CollectionProductSection = ({
...
-              <CommandBar.Command
-                action={handleRemove}
-                shortcut="r"
-                label={t("general.remove")}
-              />
-            </CommandBar.Bar>
-          </CommandBar>
-        </div>
-      )}
+      <DataTable
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/routes/collections/collection-detail/components/collection-product-section/collection-product-section.tsx:105]: comment: Dang, looking at these refactors is always a pleasure

File: packages/admin-next/dashboard/src/routes/gift-cards/gift-card-detail/loader.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import { LoaderFunctionArgs } from "react-router-dom"
+
+import { medusa, queryClient } from "../../../lib/medusa"
+
+const giftCardDetailQuery = (id: string) => ({
+  queryKey: adminProductKeys.detail(id),
+  queryFn: async () => medusa.admin.giftCards.retrieve(id),
+})
+
+export const giftCardLoader = async ({ params }: LoaderFunctionArgs) => {
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/routes/gift-cards/gift-card-detail/loader.ts:13]: almost looks _remixy_

**Review by olivermrbl — APPROVED** [review]
Left some comments, but otherwise this is very nice

File: packages/admin-next/dashboard/src/routes/gift-cards/gift-card-detail/loader.ts (hunk 3 above)
Comment (kasperkristensen) [reply packages/admin-next/dashboard/src/routes/gift-cards/gift-card-detail/loader.ts:13]: When the `unstable_` is gone here I am pulling the trigger 🔫 

https://remix.run/docs/en/main/future/vite

**Pull request merged**